pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,489 @@
+           "ACV": "F2",
+           "R2W": "F3",
+           "R4W": "F4",
+           "DCI": "F5",
+           "ACI": "F6",
+           "Rext": "F7",
+           }
+
+    RANGES={"DCV": {3E-2: "R-2", 3E-1: "R-1",3: "R0",30: "R1",300:"R2",
+                    "auto": "RA"},
+            "ACV": {3E-1: "R-1", 3: "R0",30: "R1" ,300: "R2","auto": "RA"},
+            "R2W": {30: "R1", 300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5",
+                    3E6: "R6", 3E7: "R7","auto": "RA"},
+            "R4W": {30: "R1",300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5",
+                    3E6: "R6", 3E7: "R7","auto": "RA"},
+            "DCI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "ACI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "Rext": {3E7: "R7","auto": "RA"},
+            }
+
+    TRIGGERS={
+            "auto": "T1",
+            "internal": "T1",
+            "external": "T2",
+            "single": "T3",
+            "hold": "T4",
+            "fast": "T5",
+            }
+
+    ERRORS={
+            1: "Calibration error",
+            2: "RAM error",
+            4: "ROM error",
+            8: "AD slope error",
+            16: "AD converter error",
+            32: "AD link error",
+           }
+    
+
+
+    #decoder functions
+    @classmethod
+    def decode_status(self,status_bytes, field=None):
+        """Method to handle the decoding of the status bytes into something meaningfull
+
+        :param status_bytes:    list of bytes to be decoded
+        :return ret_val:
+
+        """
+        ret_val = Status(Status_bytes(*status_bytes))
+        if field is None:
+            return ret_val.b
+        elif field == "SRQ":
+            return getattr(ret_val.B, "byte3")
+        else:
+            return getattr(ret_val.b, field)
+
+    @classmethod
+    def decode_mode(cls,status_bytes):
+        """Method to decode current mode
+
+        :param status_bytes:   list of bytes to be decoded
+        :return cur_mode: string with the current trigger mode
+        :rtype cur_mode: str
+
+        """
+        cur_stat = Status(Status_bytes(*status_bytes))
+        function = cur_stat.b.function
+        inv_modes = {v: k for k, v in HP3478A.MODES.items()}
+        cur_mode = inv_modes["F"+str(function)]
+        return cur_mode
+
+    @classmethod
+    def decode_range(cls,status_bytes):
+        """Method to decode current range
+
+        :param status_bytes:   list of bytes to be decoded
+        :return cur_range: string with the current trigger mode
+        :rtype cur_range: float
+
+        """
+        cur_stat = Status(Status_bytes(*status_bytes))
+        function = cur_stat.b.function
+        inv_modes = {v: k for k, v in HP3478A.MODES.items()}
+        cur_mode = inv_modes["F"+str(function)]
+        rnge = cur_stat.b.range
+        cur_range = list(HP3478A.RANGES[cur_mode].keys())[rnge-1]
+        return cur_range
+
+    @classmethod
+    def decode_trigger(cls,status_bytes):
+        """Method to decode trigger mode
```

Comment:
If you replace `@classmethod` with `@staticmethod` and remove the `cls` parameter, you should be good to go with static method.